tests/ActivityLoggerTest.php

```diff
@@ -195,6 +195,36 @@ public function it_will_use_the_logged_in_user_as_the_causer_by_default()
         $this->assertEquals($userId, $this->getLastActivity()->causer->id);
     }
 
+    /** @test */
+    public function it_can_log_activity_using_an_anonymous_causer()
+    {
+        activity()
+            ->causedByAnonymous()
+            ->log('hello poetsvrouwman');
+
+        $this->assertNull($this->getLastActivity()->causer_id);
+        $this->assertNull($this->getLastActivity()->causer_type);
+    }
+
+    /**
+     * @test
+     *
+     * @requires !Travis
+     */
+    public function it_will_override_the_logged_in_user_as_the_causer_when_an_anonymous_causer_is_specified()
+    {
+        $userId = 1;
+
+        Auth::login(User::find($userId));
+
+        activity()
+            ->causedByAnonymous()
+            ->log('hello poetsvrouwman');
+
+        $this->assertEquals(null, $this->getLastActivity()->causer_id);
```

Comment:
Sorry, haven't noted it: here also `assertNull()` please.

Reply:
https://github.com/spatie/laravel-activitylog/pull/605/commits/9c6f40c2f5bde39c9d7f221f0c0fbd488184e2a5